SmartDeviceLink/SDLLockScreenPresenter.m

```diff
@@ -47,30 +45,41 @@ - (void)stop {
     }
 
     // Dismiss and destroy the lockscreen window
-    [self sdl_dismissWithCompletionHandler:^{
-        self.lockWindow = nil;
+    [self sdl_dismissWithCompletionHandler:^(BOOL success) {
+        if (success) {
+            self.lockWindow = nil;
+            self.lockViewController = nil;
+        }
     }];
 }
 
 /// Shows or hides the lockscreen with an animation. If the lockscreen is shown/dismissed in rapid succession the final state of the lockscreen may not match the expected state as the order in which the animations finish can be random. To guard against this scenario, store the expected state of the lockscreen. When the animation finishes, check the expected state to make sure that the final state of the lockscreen matches the expected state. If not, perform a final animation to the expected state.
 /// @param show True if the lockscreen should be shown; false if it should be dismissed
-- (void)updateLockScreenToShow:(BOOL)show {
+- (void)updateLockScreenToShow:(BOOL)show withCompletionHandler:(nullable SDLLockScreenDidFinishHandler)completionHandler {
     // Store the expected state of the lockscreen
     self.shouldShowLockScreen = show;
 
     if (show) {
         [self sdl_presentWithCompletionHandler:^{
-            if (self.shouldShowLockScreen) { return; }
+            if (self.shouldShowLockScreen) {
+                if (completionHandler != nil) { completionHandler(); }
+                return;
+            }
 
             SDLLogV(@"The lockscreen has been presented but needs to be dismissed");
-            [self sdl_dismissWithCompletionHandler:nil];
+            [self sdl_dismissWithCompletionHandler:^(BOOL success) {
+                completionHandler();
```

Comment:
The app is crashing with a threading error here:

Branch installed on LMP and connected to SYNC 3.0 head unit.
1. Set the SYNC v3.0 head unit to "parked"
1. Open LMP and connect the device to the head unit using a USB cord.
1. Launch the SDL app by selecting the icon on the HMI.

### Expected result
Lock screen shows  and dismisses

### Actual result
Lock screen shows and then the app crashes.

Reply:
I'm not seeing this crash either on the example app or LMP.

Reply:
I just had it crash twice here. I think the issue is that `completionHandler` is `nil`. Should probably check if it is `nil` before trying to call it.

Reply:
I think I fixed it.